lib/request-handler.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} requestHandler
+ * @function buildRequestOptions
+ * @function checkResponse
+ * @function createUrl
+ */
+
+
+/**
+ * Creates requestHandler.
+ *
+ * @param {bluebird} bluebird
+ * @param {dateService} dateService
+ * @param {string} host
+ * @param {logger} logger
+ * @param {urijs} Uri
+ * @return {requestHandler}
+ */
+module.exports = (bluebird, dateService, host, logger, Uri) => {
+    // This error object is a last ditch attempt to supply info if
+    // OpenToken does not provide a body.
```

Comment:
I'm concerned with this particular comment.  If OpenToken does not provide a body, do you really want to fabricate a body?